controllers/api/favorite-pwa.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const express = require('express');
+const router = express.Router(); // eslint-disable-line new-cap
+
+const verifyIdToken = require('../../lib/verify-id-token');
+const libFavoritePwa = require('../../lib/favorite-pwa');
+const FavoritePwa = require('../../models/favorite-pwa');
+const User = require('../../models/user');
+
+/**
+ * GET /favorite-pwa/
+ *
+ * Returns all Favorite PWAs for a user
+ */
+router.get('/', (req, res) => {
+  res.setHeader('Content-Type', 'application/json');
+
+  const idToken = req.query.idToken;
```

Comment:
Sending the token as a POST parameter (as in "Add PWA") is better than a GET query string. (If it's in the query string, it's susceptible to being logged, saved to browser history, leaked to third parties via the referrer, etc.)

I think the way the "star" function should work is via an XHR/fetch() POST to some endpoint.  Currently the code [automatically sets the `idToken` form element](https://github.com/GoogleChrome/gulliver/blob/master/public/js/gulliver.es6.js#L78) when a page is loaded, or whenever the user changes. (If they switch accounts, for example.)

It could be enhanced to set `document.idToken` or similar for easy access to the starring code. (Or maybe create an `authenticatedFetch()` function that works like `fetch()`, except that it injects the `Authorization` header if available. This would also be a good point to add CSRF handling, if we do that.)